src/NUnitFramework/framework/Constraints/CollectionTally.cs

```diff
@@ -106,5 +91,26 @@ public void TryRemove(IEnumerable c)
             foreach (object o in c)
                 TryRemove(o);
         }
+
+        private class TallyEqualityComparer : EqualityComparer<object>
+        {
+            private readonly NUnitEqualityComparer _comparer;
+            private Tolerance _tolerance = Tolerance.Default;
+
+            public TallyEqualityComparer(NUnitEqualityComparer comparer)
+            {
+                _comparer = comparer;
+            }
+
+            public override bool Equals(object x, object y)
+            {
+                return _comparer.AreEqual(x, y, ref _tolerance);
+            }
+
+            public override int GetHashCode(object obj)
+            {
+                return Default.GetHashCode(obj);
```

Comment:
@ggeurts There are two problems with this GetHashCode implementation.

The first is that it does not account for things like `.IgnoreCase` and `.Using(myCustomComparer)`. This means it breaks the IEqualityComparer contract: if `Equals(x, y)` is true, then GetHashCode(x) *must* equal `GetHashCode(y)`. When this contract is broken, hash tables like Dictionary which rely on this will give false negatives. This will cause bad results.

The second problem is deeper. We can't use `IEqualityComparer` at all if `NUnitEqualityComparer` has a nonzero tolerance. It'll give broken results because the hashcode for that is incorrect. We'll have to fall back to the old behavior if there is a tolerance. Were you following the discussion where we were talking about tolerances and transitivity?

Reply:
@ggeurts If writing correct, custom GetHashCode methods for every `NUnit.Framework.Constraints.Comparers` comparer was not what you had in mind when you started this PR, that's on me. I apologize for failing to lay out more concrete expectations. If you aren't interested, don't feel bad. If you are interested, I'm happy to help.